packages/@react-stately/menu/src/useMenuState.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {Key, useCallback, useMemo, useState} from 'react';
+import type {MenuTriggerState} from './useMenuTriggerState';
+
+export interface MenuTreeState {
+  // TODO descriptions
+  expandedKeysStack: Key[],
+  closeAll: () => void,
+  openSubMenu: (triggerKey: Key, level: number) => void,
+  closeSubMenu: (triggerKey: Key, level: number) => void
+}
+
+// TODO descriptions and naming. useMenuState or useMenuTreeState?
+// Added props here so we can handle adding props in the future if need be, otherwise it would be a breaking change
+export function UNSTABLE_useMenuState(props: {}, state: MenuTriggerState): MenuTreeState  {
```

Comment:
open to opinions, also added placeholder prop arg for future additions

Reply:
TODO: move this stuff into useMenuTriggerState and add UNSTABLE to the return stuff